python/ray/data/preprocessors/batch_mapper.py

```diff
@@ -36,19 +39,53 @@ class BatchMapper(Preprocessor):
         >>> preprocessor = BatchMapper(fn)
         >>> preprocessor.transform(ds)  # doctest: +SKIP
         Dataset(num_blocks=1, num_rows=3, schema={X: int64})
+        >>>
+        >>> def fn_numpy(batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:
+        ...     return {"X": batch["X"]}
+        >>> preprocessor = BatchMapper(fn_numpy, batch_format="numpy")
+        >>> preprocessor.transform(ds)  # doctest: +SKIP
+        Dataset(num_blocks=1, num_rows=3, schema={X: int64})
 
     Args:
         fn: The function to apply to data batches.
+        batch_format: The preferred batch format to use in UDF. If not given,
+            default to pandas.
     """
 
     _is_fittable = False
 
-    def __init__(self, fn: Callable[["pandas.DataFrame"], "pandas.DataFrame"]):
+    def __init__(
+        self,
+        fn: Callable[
+            [Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]]],
+            Union["pandas.DataFrame", np.ndarray, Dict[str, np.ndarray]],
+        ],
+        batch_format="pandas",
+        # TODO: We should reach consistency of args between BatchMapper and map_batches.
+    ):
+        if batch_format not in [
+            "pandas",
+            "numpy",
+        ]:
+            raise ValueError("BatchMapper only supports pandas and numpy batch format.")
+
+        self.batch_format = batch_format
         self.fn = fn
 
+    def _transform_numpy(
+        self, np_data: Union[np.ndarray, Dict[str, np.ndarray]]
+    ) -> Union[np.ndarray, Dict[str, np.ndarray]]:
+        return self.fn(np_data)
+
     def _transform_pandas(self, df: "pandas.DataFrame") -> "pandas.DataFrame":
         return self.fn(df)
 
+    def _determine_transform_to_use(self, data_format: str):
+        if self.batch_format:
+            return self.batch_format
+        else:
+            return super()._determine_transform_to_use(data_format)
```

Comment:
If batch_format is not allowed to be None then we never go inside this else statement.